src/views/list_view.rs

```diff
@@ -222,6 +229,43 @@ impl ListView {
             .max()
             .unwrap_or(0)
     }
+    fn labels_size(&self) -> ColumnSize {
+        let mut labels_size = ColumnSize {max_width: 0, sum_height: 0};
+        let mut scan = self
+            .children
+            .iter()
+            .map(ListChild::label)
+            .scan(&mut labels_size, |s, label| {
+                let width = label.width();
+                if width > (**s).max_width {
+                    (**s).max_width = width;
+                }
+                (**s).sum_height += 1;
+                //println!("{:#?}", **s);
+                Some((**s).clone())
+            });
+        while scan.next().is_some() {}
+        return labels_size;
+    }
+    fn views_size(&mut self, req: Vec2) -> ColumnSize {
+        let mut views_size = ColumnSize {max_width: 0, sum_height: 0};
+        let mut scan = self
+            .children
+            .iter_mut()
+            .filter_map(ListChild::view)
+            .map(|v| v.required_size(req))
+            .scan(&mut views_size, |s, size| {
+                let width = size.x;
+                if width > (**s).max_width {
+                    (**s).max_width = width;
+                }
+                (**s).sum_height += size.y;
+                println!("view_size: {:#?}", **s);
+                Some((**s).clone())
+            });
+        while scan.next().is_some() {}
```

Comment:
Not sure if that's really the **rust way™️**

Reply:
Could be `for _ in scan {}`?
But I think rather than the `.scan()`, a `for`-loop may be clearer.

```rust
        let mut labels_size = ColumSize { max_width: 0, max_height: 0 };
        for label in self.children.iter().map(ListChild::label) {
                let width = label.width();
                if width > labels_size.max_width {
                    labels_size.max_width = width;
                }
                labels_size.sum_height += 1;
        }
```

Or even simpler:

```rust
        // Convenient function to take the max of values on one axis, and the sum on the other axis
        let max_size = cursive::direction::Orientation::Vertical.stack(
            self.children.iter().map(|c| Vec2::new(c.label().width(), 1))
        );

        // Then it's easy to convert to a `ColumnSize` - but why even need a `ColumnSize` type vs `Vec2`?
        let labels_size = ColumnSize { max_width: max_size.x, max_height: max_size.y };
```